### PR TITLE
re-commiting

### LINKED ISSUE #107: Basic Look & Room Description
**Problem**  
Player needs initial spatial awareness after startup.

**Proposed solution**  
Create `ExplorationService` and `DefaultExplorationService.explore()` + `describeCurrentRoom()` to print room name, description, items, exits, and any alive monster.

**Acceptance criteria**  
- [ ] Implement service + wire `MovementController` to handle `explore`
- [ ] Invalid current room ID → friendly failure message
- [ ] Unit test with a small in-memory world

**Notes/risks**  
Ensure consistent newline formatting.

### LINKED ISSUE #108: Movement Between Rooms
**Problem**  
Player navigates using directions; movement must validate exits.

**Proposed solution**  
Add `DefaultExplorationService.move()` to change `player.roomId` if exit exists; handle invalid directions and missing exits.

**Acceptance criteria**
- [ ] `go <dir>` updates player location on valid exits; otherwise returns clear failure
- [ ] Support direction aliases (n/s/e/w → canonical)
- [ ] `MovementController` routes `GO` with direction argument
- [ ] Unit tests for valid/invalid movement

**Notes/risks**  
Puzzle presentation will be added later when the interaction service exists.

### LINKED ISSUE #112: Puzzle Presentation & Solving
**Problem**  
Rooms with puzzles must present a prompt and allow solving attempts with limited retries.

**Proposed solution**  
Create `InteractionService` and `DefaultInteractionService` for `presentPuzzle()` on room entry and `solve <answer>` with attempts/lockout and optional reward item.

**Acceptance criteria**  
- [ ] Entering a puzzle room presents prompt and sets an awaiting-answer flag in context
- [ ] Correct answer → phase SOLVED; reward item added if configured
- [ ] Wrong answer decrements attempts; lockout on zero; leaving room resets puzzle to LOCKED
- [ ] `InteractionController` wires `solve`; unit tests for success/failure/reset

### LINKED ISSUE #113: Integrate Puzzle Presentation with Movement
**Problem**  
Puzzle presentation should occur upon entering rooms that have puzzles (or as specified by SRS for other specific puzzles).

**Proposed solution**  
Update `DefaultExplorationService.move()` to call `presentPuzzle(...)` after successful movement.

**Acceptance criteria**  
- [ ] Movement into a puzzle room presents the puzzle; movement tests updated
- [ ] Exiting an in-progress puzzle room resets puzzle state

**Notes/risks**  
Ensure clean separation between services.

### LINKED ISSUE #114: Combat: Initiate / Attack / Ignore / Death Handling
**Problem**  
Player encounters monsters and may fight or ignore them.

**Proposed solution**  
Create `CombatService` + `DefaultCombatService` to manage combat rounds, critical hit chance, and death handling; store combat state in `GameContext`.

**Acceptance criteria**  
- [ ] `attack <monster>` starts combat if alive and name matches; alternate player/monster turns
- [ ] Deterministic tests via seeded RNG for critical hits
- [ ] Death handling prints game over and options; `ignore` removes monster
- [ ] `CombatController` wires `attack/ignore`; unit tests for edge cases
- [ ] Handle loot rewards after monster is killed

### PR DESCRIPTION
**Pull request by dmusau23**
##Recommitting the branch

### CODE REVIEW

File: game/src/main/java/avengers/client/controller/InteractionController.java [hunk 1]
```diff
@@ -1,20 +1,198 @@
...
+            .map(Optional::get)
+            .filter(p -> !p.isSolved())
+            .findFirst();
+
+    if (puzzleOpt.isEmpty()) {
+      return CommandResult.fail("All puzzles in this room are solved.");
+    }
+
+    return CommandResult.success(
+        "\n💡 Hint: Read the puzzle description carefully. The answer is often hidden in plain sight.\n");
```
Comment (dpandev) [inline game/src/main/java/avengers/client/controller/InteractionController.java:122]: @dmusau23 this emoji can be removed

File: game/src/main/java/avengers/domain/model/Puzzle.java [hunk 2]
```diff
@@ -7,6 +7,7 @@
  * specific solution, reward for success, and consequence for failure.
  */
 public final class Puzzle {
+  //
```
Comment (dpandev) [inline game/src/main/java/avengers/domain/model/Puzzle.java:10]: can be removed

File: game/src/main/java/avengers/client/controller/SystemController.java [hunk 3]
```diff
@@ -16,6 +16,7 @@ public class SystemController implements CommandController {
   public SystemController(SaveService save, WorldLoader worldLoader) {
     this.save = save;
     this.worldLoader = worldLoader;
+    //
```
Comment (dpandev) [inline game/src/main/java/avengers/client/controller/SystemController.java:19]: can be removed

File: game/src/main/java/avengers/client/controller/GameController.java [hunk 4]
```diff
@@ -13,6 +13,7 @@
  * Example controller class. Replace with actual controllers (GameController, MenuController, etc.)
  */
 public final class GameController {
+  //
```
Comment (dpandev) [inline game/src/main/java/avengers/client/controller/GameController.java:16]: can be removed

File: game/src/main/java/avengers/client/controller/CommandController.java [hunk 5]
```diff
@@ -10,6 +10,7 @@
  * they support and how to handle them.
  */
 public interface CommandController {
+  //
```
Comment (dpandev) [inline game/src/main/java/avengers/client/controller/CommandController.java:13]: can be removed

File: game/src/main/java/avengers/client/command/InputCommandParser.java [hunk 6]
```diff
@@ -55,6 +55,7 @@ public CommandToken parse(final String line) {
       return new CommandToken(Verb.GO, directionSynonyms.get(head), List.of(), raw);
     }
 
+    //
```
Comment (dpandev) [inline game/src/main/java/avengers/client/command/InputCommandParser.java:58]: can be removed

File: game/src/main/java/avengers/service/SaveService.java [hunk 7]
```diff
@@ -14,6 +14,7 @@
 
 /** Service responsible for saving and loading game data. */
 public final class SaveService {
+  //
```
Comment (dpandev) [inline game/src/main/java/avengers/service/SaveService.java:17]: can be removed

File: game/src/main/java/avengers/service/world/JsonWorldLoader.java [hunk 8]
```diff
@@ -143,6 +143,8 @@ private Map<String, Puzzle> loadPuzzles() {
     return new HashMap<>();
   }
 
+  //
```
Comment (dpandev) [inline game/src/main/java/avengers/service/world/JsonWorldLoader.java:146]: can be removed

File: game/src/main/java/avengers/service/DefaultInteractionService.java
Comment (dpandev) [inline game/src/main/java/avengers/service/DefaultInteractionService.java:1]: Some of these Puzzle methods do not exist in the Puzzle class. Will need to refactor this class to use the methods existing in the Puzzle class and/or refactor the Puzzle class to include additional classes if needed.

**Review by dpandev — CHANGES_REQUESTED** [review]
Some refactoring is needed
